test_runner/src/main/kotlin/ftl/shard/Shard.kt

```diff
@@ -136,7 +136,8 @@ object Shard {
         testcases.sortByDescending { it.time }
 
         testcases.forEach { testMethod ->
-            val shard = shards.first()
+            // num_shards must be > 1, and <= 50
+            val shard = shards.first { it.testMethods.size + args.testTargetsAlwaysRun.size < 50 }
```

Comment:
There is a problem here. According to doc the number of shards should be less than 50, but this condition prevents from putting more than 50 test methods per shard so probably it's not what we want.